### PR TITLE
Enable Travis CI build for test and samples

### PR DESCRIPTION
**Pull request by mocabe**
This PR adds script for building samples and tests on Travis CI

Hi, I just did quick experiment to enable CI build for tests and samples.
Current build includes compilers like:
* `g++` 6, 7, 8
* `clang++` 3.6, 4.0, 5.0, 6.0

`g++5` cannot compile samples since old libstdc++ doesn't work well with C++14 code, so currently it is excluded from matrix.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,38 +12,128 @@ matrix:
           sources:
           - ubuntu-toolchain-r-test
           packages:
-          - gcc-5
-          - g++-5
-      env: GCC_VERSION=5
-      compiler: gcc-5
+          - g++-6
```
Comment (mtavenrath) [inline .travis.yml]: Is it possible to keep gcc-5 and skip the samples build for this compiler?

File: samples/utils/CMakeLists.txt [hunk 2]
```diff
@@ -40,4 +40,5 @@ target_link_libraries(utils PUBLIC glfw)
 target_link_libraries(utils PUBLIC glslang)
 target_link_libraries(utils PUBLIC glslang-default-resource-limits)
 target_link_libraries(utils PUBLIC SPIRV)
-target_compile_definitions(utils PUBLIC VULKAN_HPP_DISPATCH_LOADER_DYNAMIC=1)
\ No newline at end of file
+target_compile_definitions(utils PUBLIC VULKAN_HPP_DISPATCH_LOADER_DYNAMIC=1)
+target_compile_features(utils PUBLIC cxx_std_14)
```
Comment (mtavenrath) [inline samples/utils/CMakeLists.txt]: If c++14 is required for the samples travis won't find potential c++11 features.

File: samples/01_InitInstance/CMakeLists.txt [hunk 3]
```diff
@@ -14,22 +14,26 @@
...
-source_group(sources FILES ${SOURCES})
-
-add_executable(01_InitInstance
-  ${HEADERS}
-  ${SOURCES}
-)
-
-set_target_properties(01_InitInstance PROPERTIES FOLDER "Samples")
-target_link_libraries(01_InitInstance "${Vulkan_LIBRARIES}")
+endif()
```
Comment (mtavenrath) [inline samples/01_InitInstance/CMakeLists.txt]: pedantic: EOF should be a newline everywhere

**Comment by mocabe** [comment]
@mtavenrath The reason why C++14 is enabled in samples is there are already some C++14 codes in there, i.e. `std::make_unique`, `std::string_literals`, and generic lambdas.  Changes required to make it compile in C++11 mode seems relatively small, so I'll work on that.

**Pull request merged**